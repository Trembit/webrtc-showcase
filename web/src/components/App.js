/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';

const ContextType = {
  // Enables critical path CSS rendering
  // https://github.com/kriasoft/isomorphic-style-loader
  insertCss: PropTypes.func.isRequired,
  // Universal HTTP client
  fetch: PropTypes.func.isRequired,
};

/**
 * The top-level React component setting context (global) variables
 * that can be accessed from all the child components.
 *
 * https://facebook.github.io/react/docs/context.html
 *
 * Usage example:
 *
 *   const context = {
 *     history: createBrowserHistory(),
 *     store: createStore(),
 *   };
 *
 *   ReactDOM.render(
 *     <App context={context}>
 *       <Layout>
 *         <LandingPage />
 *       </Layout>
 *     </App>,
 *     container,
 *   );
 */
class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };

  static childContextTypes = ContextType;

  getChildContext() {
    return this.props.context;
  }

  render() {
    // NOTE: If you need to add or modify header, footer etc. of the app,
    // please do that inside the Layout component.
    return React.Children.only(this.props.children);
  }
}

export default App;
