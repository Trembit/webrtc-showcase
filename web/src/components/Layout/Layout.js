/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';

// external-global styles must be imported in your JS.
import normalizeCss from 'normalize.css';
import s from './Layout.css';
import Header from '../Header';
import Feedback from '../Feedback';
import Footer from '../Footer';

class Layout extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  render() {
    return (
      <div>
        <Header />
        {this.props.children}
        <Feedback />
        <Footer />
      </div>
    );
  }
}

export default withStyles(normalizeCss, s)(Layout);
