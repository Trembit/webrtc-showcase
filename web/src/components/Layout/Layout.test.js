/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Layout from './Layout';

describe('Layout', () => {
  test('renders children correctly', () => {
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {}, fetch: () => {} }}>
          <Layout>
            <div className="child" />
          </Layout>
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
