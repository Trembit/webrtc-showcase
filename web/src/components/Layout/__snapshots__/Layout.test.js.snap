// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div>
  <div
    className="root"
  >
    <div
      className="container"
    >
      <div
        className="root"
        role="navigation"
      >
        <a
          className="link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="link"
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
        <span
          className="spacer"
        >
           | 
        </span>
        <a
          className="link"
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
        <span
          className="spacer"
        >
          or
        </span>
        <a
          className="link highlight"
          href="/register"
          onClick={[Function]}
        >
          Sign up
        </a>
      </div>
      <a
        className="brand"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="React"
          height="38"
          src="logo-small.png"
          srcSet="logo-small@2x.png 2x"
          width="38"
        />
        <span
          className="brandTxt"
        >
          Your Company
        </span>
      </a>
      <div
        className="banner"
      >
        <h1
          className="bannerTitle"
        >
          React
        </h1>
        <p
          className="bannerDesc"
        >
          Complex web apps made easy
        </p>
      </div>
    </div>
  </div>
  <div
    className="child"
  />
  <div
    className="root"
  >
    <div
      className="container"
    >
      <a
        className="link"
        href="https://gitter.im/kriasoft/react-starter-kit"
      >
        Ask a question
      </a>
      <span
        className="spacer"
      >
        |
      </span>
      <a
        className="link"
        href="https://github.com/kriasoft/react-starter-kit/issues/new"
      >
        Report an issue
      </a>
    </div>
  </div>
  <div
    className="root"
  >
    <div
      className="container"
    >
      <span
        className="text"
      >
        © Your Company
      </span>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/admin"
        onClick={[Function]}
      >
        Admin
      </a>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/privacy"
        onClick={[Function]}
      >
        Privacy
      </a>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/not-found"
        onClick={[Function]}
      >
        Not Found
      </a>
    </div>
  </div>
</div>
`;
